src/bases/D3MPlanBase.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+abstract contract D3MPlanBase {
+
+        // --- Auth ---
+    mapping (address => uint256) public wards;
```

Comment:
So I'm not so sure now that the auth stuff should be kept in the base implementation. It depends on whether we imagine an automatic plan, which is probably possible.
So leaning towards moving this (wards, rely/deny, events, modifier) to the derived contract.

Reply:
Rethinking this - we might want to keep the auth stuff in the base contract as in almost all cases there will be some configurations to do, and also since otherwise need to implement it correctly and test it in multiple places.

Reply:
yeah i would keep it there even if there is the possibility it is not used in a few cases.